twisted/protocols/tls.py

```diff
@@ -321,20 +315,32 @@ def makeConnection(self, transport):
         # Now that we ourselves have a transport (initialized by the
         # ProtocolWrapper.makeConnection call above), kick off the TLS
         # handshake.
+        self._checkHandshakeStatus()
+
+
+    def _checkHandshakeStatus(self):
+        """
+        Ask OpenSSL to proceed with a handshake in progress.
 
+        Initially, this just sends the ClientHello; after some bytes have been
+        stuffed in to the C{Connection} object by C{dataReceived}, it will then
+        respond to any C{Certificate} or C{KeyExchange} messages.
+        """
         # The connection might already be aborted (eg. by a callback during
         # connection setup), so don't even bother trying to handshake in that
         # case.
-        if not self._aborted:
-            try:
-                self._tlsConnection.do_handshake()
-            except WantReadError:
-                # This is the expected case - there's no data in the
-                # connection's input buffer yet, so it won't be able to
-                # complete the whole handshake now. If this is the speak-first
-                # side of the connection, then some bytes will be in the send
-                # buffer now; flush them.
-                self._flushSendBIO()
+        if self._aborted:
+            return
+        try:
+            self._tlsConnection.do_handshake()
+        except WantReadError:
+            self._flushSendBIO()
```

Comment:
This is maybe just because I know too much about OpenSSL, but it's standing out to me that we don't handle `WantWriteError`. I think I can convince myself that we will never need it, but I'd at least like a comment to float around here that says "We don't need to handle WantWriteError because it should never be emitted."